packages/visual-group/src/group-helper/matrix-model.js

```diff
@@ -213,14 +217,15 @@ export const getMatrixModel = (dataModel, facetsAndProjections, valueCellCreator
         const field = colFacetNames[0];
         const firstLevelColumnKeys = retriveDomainFromData(dataModel, field);
 
-        sortFacetFields(colFacets[0], firstLevelColumnKeys, globalConfig);
-
         // Get unique keys to create faceted datamodels: this time for columns
         uniqueKeyGenerator(columnKeys, {
             facets: colFacetNames,
             dataModel,
             uniqueValues: firstLevelColumnKeys
         });
+
+        sortConfig && sortFacetFields(colFacets, columnKeys, globalConfig);
```

Comment:
Why the check for sortConfig is present multiple times.

A better way would be to have a default sortConfig value and use sorting accordingly instead of putting multiple checks

Reply:
Removed check with a default value for `sort`